### PR TITLE
Feat/[NRPTI-1194] - Move hardcoded Acts and Regulations to DB

### PR DESCRIPTION
**Pull request by sanjaytkbabu**
## Pull Request Standards

- [x] The title of the PR is accurate
- [x] The title includes the type of change [`HOTFIX`, `FEATURE`, `etc`]  
- [x] The PR title includes the ticket number in format of `[NRPTI-###]`
- [x] Documentation is updated to reflect change

# Description

This PR includes the following proposed change(s):

- Moved hardcoded Acts and Regulations to DB
- Added a cronjob to update Energy Resource Activities Act from API
- Added migration to create a new collection to store act-regulations mapping
- Added migration to update any existing records with Oil and Gas Activities act to Energy Resource Activities act
- Added 2 new endpoints for - updating act title through cronjob and fetching all acts and regulations from DB

Issue NOT addressed: If the Energy Resource Activity Act were updated next month, the app would automatically reflect this change. In the front end, filters and options would show the new value. For records, all records that get added/updated through the daily cronjob and by using the IMPORT buttons will have their values automatically updated. BUT other records associated with this act that were added from other sources like through the "add record" feature will not be updated. Then, if searching for records and filtering by the new act, these records will not appear. 

Future work should modify all existing records to replace their legislation with constant codes that map to the appropriate value. This value is then what gets updated. This way, records themselves will not need to be updated. The front end will translate the codes based on their mapped association. Implementing this feature in this ticket would have ballooned an already too large task.

### CODE REVIEW

**Review by davidclaveau — DISMISSED** [review]

File: angular/projects/admin-nrpti/src/app/services/acts.service.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  /**
+   * @constructor
+   * @param {ConfigService} configService - The configuration service.
+   * @param {HttpClient} http - The HTTP client for making API requests.
+   */
+  constructor(private configService: ConfigService, public http: HttpClient) {}
+
+  /**
+   * Initialize the service by setting the API endpoint and refreshing agencies.
+   * @asyncx
```
Comment (LocalNewsTV) [inline angular/projects/admin-nrpti/src/app/services/acts.service.ts:31]:
```suggestion
   * @async
```

File: api/src/controllers/acts-regulations-controller.js [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    { _schemaName: "ActsRegulationsMapping", actCode: actCode},
+    { $set: { actName : actTitle } }
+  );
+  console.log('updated actCode - ' + actCode + 'actTitle - ' + actTitle);
+  }
+  } catch (error) {
+    console.error("updateTitlesInDB: Failed to update DB:", error);
+  }
+
+}
```
Comment (LocalNewsTV) [inline api/src/controllers/acts-regulations-controller.js:57]: Indentation

File: api/src/integrations/bcogc/inspections-utils.js [hunk 3]
```diff
@@ -27,15 +27,15 @@ class Inspections extends BaseRecordUtils {
...
    * @returns a inspection object matching the format expected by the API record post/put controllers.
    * @memberof Inspections
    */
-  transformRecord(csvRow) {
+  transformRecord(csvRow, actName) {
     if (!csvRow) {
       throw Error('transformRecord - required csvRow must be non-null.');
     }
-
+
```
Comment (LocalNewsTV) [inline api/src/integrations/bcogc/inspections-utils.js:38]:
```suggestion

```

**Review by LocalNewsTV — COMMENTED** [review]
my Angular knowledge is weak, so my review is not a comprehensive as it should be. Someone else should give final approval and troubleshoot.

File: angular/projects/global/src/lib/utils/act-data-service-nrpti.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+          } else {
+            regulations[regulation] = [act];
+          }
+        })
+      );
+
+      return regulations;
+    }
+  };
+}
```
Comment (LocalNewsTV) [inline angular/projects/global/src/lib/utils/act-data-service-nrpti.ts:61]: This function is double nested. It was flagged by Sonarcloud

Comment (acatchpole) [reply angular/projects/global/src/lib/utils/act-data-service-nrpti.ts]: fixed

File: angular/projects/common/src/app/utils/record-constants.ts [hunk 5]
```diff
@@ -265,548 +288,17 @@ export class Picklists {
...
    * Returns an array of all supported legislation act strings.
    *
    * @static
    * @memberof Picklists
    * @returns {string[]} sorted array of acts
    */
-  public static getAllActs = function(): string[] {
-    return Object.keys(this.legislationActsMappedToRegulations).sort((a, b) => a.localeCompare(b));
-  };
+  // public static getAllActs = function(): string[] {
```
Comment (fergmac) [inline angular/projects/common/src/app/utils/record-constants.ts:299]: Commented code to be removed?

**Review by fergmac — APPROVED** [review]

**Pull request merged**